### ISSUE TITLE
Crunchyroll `Unauthenticated request`

### ISSUE THREAD

**Issue by nih0** [chrippa/livestreamer#872]
Is it me just being stupid?

```python
 ~ livestreamer http://www.crunchyroll.com/gintama/episode-266-you-can-never-pause-at-the-perfect-time-678765 --crunchyroll-username nih0_
Enter Crunchyroll password:
[cli][info] Found matching plugin crunchyroll for URL http://www.crunchyroll.com/gintama/episode-266-you-can-never-pause-at-the-perfect-time-678765
Traceback (most recent call last):
  File "/usr/local/bin/livestreamer", line 9, in <module>
    load_entry_point('livestreamer==1.12.1', 'console_scripts', 'livestreamer')()
  File "/Library/Python/2.7/site-packages/livestreamer_cli/main.py", line 886, in main
    handle_url()
  File "/Library/Python/2.7/site-packages/livestreamer_cli/main.py", line 471, in handle_url
    streams = fetch_streams(plugin)
  File "/Library/Python/2.7/site-packages/livestreamer_cli/main.py", line 388, in fetch_streams
    sorting_excludes=args.stream_sorting_excludes)
  File "/Library/Python/2.7/site-packages/livestreamer/plugin/plugin.py", line 313, in get_streams
    return self.streams(*args, **kwargs)
  File "/Library/Python/2.7/site-packages/livestreamer/plugin/plugin.py", line 227, in streams
    ostreams = self._get_streams()
  File "/Library/Python/2.7/site-packages/livestreamer/plugins/crunchyroll.py", line 211, in _get_streams
    api = self._create_api()
  File "/Library/Python/2.7/site-packages/livestreamer/plugins/crunchyroll.py", line 266, in _create_api
    raise err
crunchyroll.CrunchyrollAPIError: Unauthenticated request


 ~ livestreamer http://www.crunchyroll.com/gintama/episode-266-you-can-never-pause-at-the-perfect-time-678765
[cli][info] Found matching plugin crunchyroll for URL http://www.crunchyroll.com/gintama/episode-266-you-can-never-pause-at-the-perfect-time-678765
Traceback (most recent call last):
  File "/usr/local/bin/livestreamer", line 9, in <module>
    load_entry_point('livestreamer==1.12.1', 'console_scripts', 'livestreamer')()
  File "/Library/Python/2.7/site-packages/livestreamer_cli/main.py", line 886, in main
    handle_url()
  File "/Library/Python/2.7/site-packages/livestreamer_cli/main.py", line 471, in handle_url
    streams = fetch_streams(plugin)
  File "/Library/Python/2.7/site-packages/livestreamer_cli/main.py", line 388, in fetch_streams
    sorting_excludes=args.stream_sorting_excludes)
  File "/Library/Python/2.7/site-packages/livestreamer/plugin/plugin.py", line 313, in get_streams
    return self.streams(*args, **kwargs)
  File "/Library/Python/2.7/site-packages/livestreamer/plugin/plugin.py", line 227, in streams
    ostreams = self._get_streams()
  File "/Library/Python/2.7/site-packages/livestreamer/plugins/crunchyroll.py", line 211, in _get_streams
    api = self._create_api()
  File "/Library/Python/2.7/site-packages/livestreamer/plugins/crunchyroll.py", line 266, in _create_api
    raise err
crunchyroll.CrunchyrollAPIError: Unauthenticated request
```

**Comment by nih0** [comment]
https://github.com/chrippa/livestreamer/issues/696 same issue, do i have to escape something?

**Comment by kg** [comment]
I'm hitting this too. Tried the obvious stuff (nuking the cached device id, purging session, etc)

**Comment by shynil** [comment]
I had a similar problem.
>Traceback (most recent call last):
  File "/usr/bin/livestreamer", line 9, in <module>
    load_entry_point('livestreamer==1.12.1', 'console_scripts', 'livestreamer')()
  File "/usr/share/livestreamer/livestreamer_cli/main.py", line 852, in main
    handle_url()
  File "/usr/share/livestreamer/livestreamer_cli/main.py", line 471, in handle_url
    streams = fetch_streams(plugin)
  File "/usr/share/livestreamer/livestreamer_cli/main.py", line 388, in fetch_streams
    sorting_excludes=args.stream_sorting_excludes)
  File "/usr/lib/python2.7/dist-packages/livestreamer/plugin/plugin.py", line 313, in get_streams
    return self.streams(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/livestreamer/plugin/plugin.py", line 227, in streams
    ostreams = self._get_streams()
  File "/usr/lib/python2.7/dist-packages/livestreamer/plugins/crunchyroll.py", line 211, in _get_streams
    api = self._create_api()
  File "/usr/lib/python2.7/dist-packages/livestreamer/plugins/crunchyroll.py", line 266, in _create_api
    raise err
crunchyroll.CrunchyrollAPIError: Unauthenticated request

**Comment by chrippa** [comment]
I believe this should be fixed with #865. Please try the latest [nightly](http://docs.livestreamer.io/install.html#nightly-build)/[git](http://docs.livestreamer.io/install.html#source-code) version.

**Comment by chrippa** [comment]
I've pushed a 1.12.2 release with the fix now.
